python/variants/core.py

```diff
@@ -0,0 +1,98 @@
+import sys
+
+from pyspark.sql import SQLContext
+
+
+class VariantsContext(object):
+    """The main entry point for VariantSpark functionality.
+
+    :param ss: SparkSession
+    :type sc: :class:`.pyspark.SparkSession`
+
+    """
+
+    def __init__(self, ss=None):
+        self.sc = ss.sparkContext
+        self.sql = SQLContext.getOrCreate(self.sc)
+        self._jsql = self.sql._jsqlContext
+        self._jvm = self.sc._jvm
+        self._vs_api = getattr(self._jvm, 'au.csiro.variantspark.api')
+        jss = ss._jsparkSession
+        self._jvsc = self._vs_api.VSContext.apply(jss)
+
+        sys.stderr.write('Running on Apache Spark version {}\n'.format(self.sc.version))
+        if self.sc._jsc.sc().uiWebUrl().isDefined():
+            sys.stderr.write('SparkUI available at {}\n'.format(self.sc._jsc.sc().uiWebUrl().get()))
+        sys.stderr.write(
+            'Welcome to\n'
+            ' _    __           _             __  _____                  __    \n'
+            '| |  / /___ ______(_)___ _____  / /_/ ___/____  ____ ______/ /__  \n'
+            '| | / / __ `/ ___/ / __ `/ __ \/ __/\__ \/ __ \/ __ `/ ___/ //_/  \n'
+            '| |/ / /_/ / /  / / /_/ / / / / /_ ___/ / /_/ / /_/ / /  / ,<     \n'
+            '|___/\__,_/_/  /_/\__,_/_/ /_/\__//____/ .___/\__,_/_/  /_/|_|    \n'
+            '                                      /_/                         \n')
+
+    def import_vcf(self, vcf_file_path):
+        """ Shut down the VariantsContext.
+        """
+        return VCFFeatureSource(self._jvm, self._vs_api,
+                                self._jsql, self._jvsc.importVCF(vcf_file_path))
+
+    def load_label(self, label_file_path, col_name):
+        """ Loads the label source file
+
+        :param label_file_path: The file path for the label source file
+        :param the name of the column containing labels
+        """
+        return self._jvsc.loadLabel(label_file_path, col_name)
+
+    def stop(self):
+        """ Shut down the VariantsContext.
+        """
+
+        self.sc.stop()
+        self.sc = None
+
+
+class VCFFeatureSource(object):
```

Comment:
I would suggest that we create a generic python FeatureSource that can wrap any Scala FeatureSource subclass.